dcm4che-core/src/main/java/org/dcm4che3/io/DicomInputStream.java

```diff
@@ -760,27 +745,30 @@ private void readFragments(Attributes attrs, int fragsTag, VR vr)
     }
 
     public byte[] readValue() throws IOException {
-        int valLen = length;
+
         try {
-            if (valLen < 0)
-                throw new EOFException(); // assume InputStream length < 2 GiB
-            int allocLen = allocateLimit >= 0
-                    ? Math.min(valLen, allocateLimit)
-                    : valLen;
-            byte[] value = new byte[allocLen];
-            readFully(value, 0, allocLen);
-            while (allocLen < valLen) {
-                int newLength = Math.min(valLen, allocLen << 1);
-                value = Arrays.copyOf(value, newLength);
-                readFully(value, allocLen, newLength - allocLen);
-                allocLen = newLength;
+
+            if (length < 0) {
+                throw new EOFException("DicomInputStream length is less than zero");
+            }
+
+            if (length > allocateLimit) {
+                LOG.warn("Expected value length {} exceeds allocate limit {}", length, allocateLimit);
             }
+
+            byte[] value = new byte[length];
+
+            readFully(value, 0, length);
+
             return value;
+
         } catch (IOException e) {
-            LOG.warn("IOException during read of {} #{} @ {}",
-                    TagUtils.toString(tag), length, tagPos, e);
-            throw e;
+
+            throw new IOException(String.format("IOException during read of %s #%d @ %d",
+                    TagUtils.toString(tag), length, tagPos), e);
```

Comment:
I would prefer the original exception (or its cause) to be thrown instead of wrapping the exception with another IOException.

Reply:
Ok, I've updated it.